### ISSUE TITLE
security issue with xerces:xercesImpl:2.8.0 dependency

### ISSUE THREAD

**Issue by uli-f** [cdk/cdk#861]
There is a security issue with the xerces:xercesImpl version 2.8.0 that is pulled in as a dependency. This is fixed in 2.12.0:
https://mvnrepository.com/artifact/xerces/xercesImpl

**Comment by egonw** [comment]
Where did you find that? In the root `pom.xml` I find:

```xml
            <dependency>
                <groupId>xerces</groupId>
                <artifactId>xercesImpl</artifactId>
                <version>2.12.2</version>
            </dependency>
```

**Comment by uli-f** [comment]
https://github.com/cdk/cdk/blob/cdk-2.7.1/storage/io/pom.xml

See also here:
https://mvnrepository.com/artifact/org.openscience.cdk/cdk-io/2.7.1

**Comment by egonw** [comment]
Ah, got it! It was fixed in master, so missed that. Can you use 2.8.0-SNAPSHOT?

**Comment by uli-f** [comment]
I don't want to use SNAPSHOT versions due to the 'silent updates' that entails.

But I already declared a constraint in my build.gradle:

```
dependencies {
  ....
  constraints {
    implementation('xerces:xercesImpl:2.12.2') {
      because 'previous versions suffer from a vulnerability'
    }
  }
}  
```

So personally, I am fine. Just wanted to let you guys know 🤓

**Comment by egonw** [comment]
ping @johnmay, what's the release schedule?

**Comment by johnmay** [comment]
Likewise with Jena, my view is the fix here is to not use them. That might be a lot of work though

**Comment by egonw** [comment]
> Likewise with Jena, my view is the fix here is to not use them. That might be a lot of work though

yes, this is a good point. Xerces is for the CDK an optional and runtime dependency: you can use it for parsing XML files, but we support two other frameworks for that too.
